src/StreamJsonRpc.Tests/JsonRpcClient20InteropTests.cs

```diff
@@ -149,6 +150,147 @@ public async Task InvokeWithParameterPassedAsObjectAsync_NoParameter()
         Assert.Null(request["params"]);
     }
 
+    [Fact]
+    public async Task InvokeWithProgressParameterAsArray()
+    {
+        int sum = 0;
+        ProgressWithCompletion<int> progress = new ProgressWithCompletion<int>(report =>
+        {
+            sum += report;
+        });
+
+        int n = 3;
+        Task<int> invokeTask = this.clientRpc.InvokeAsync<int>("test", new object[] { n, progress });
+
+        JToken request = await this.ReceiveAsync();
+
+        long progressID = request["params"][1].Value<long>();
```

Comment:
>.Value<long>() [](start = 46, length = 14)

From a test perspective we don't care if the token is a long or some other JToken. So in the interest of writing resilient tests that fail when the product is faulty but passes when the product changes details that shouldn't matter outside, I suggest you skip the long conversion and store as a JToken. You can directly inject that JToken into your string below and Newtonsoft will serialize it as JSON for you.